app/controllers/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UsersController < ApplicationController
+
+  get '/signup' do 
+    if !session[:user_id]
+      erb :"users/create_user"
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == "" 
+      redirect "/signup"
+    else
+      @user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+      @user.save
+      session[:user_id] = @user.id
+      redirect "/tweets"
+    end
+  end
+
+  get '/login' do
+    if !logged_in?
+       erb :'/users/login'
+    else
+       redirect to '/tweets'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(:username => params[:username])
```

Comment:
I know how to validate this, but I can't seem to get the error message to pop up without using rack flash.  Is there an easier way to implement the error message?